contest/contest_info.py

```diff
@@ -121,4 +126,28 @@ def can_ask(user,contest):
     return  user_is_contestant | user_is_owner_coowner
 
 def can_reply(user,contest):
-    return has_contest_ownership(user,contest)
+    return has_contest_ownership(user,contest) 
+
+def get_contest_or_404(contest_id):
+    try:
+        contest = Contest.objects.get(id = contest_id)
+        return contest
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not register contest %s! Contest not found!' % contest_id)
+        raise Http404('Can not register contest %s! Contest not found!' % contest_id)
+
+def can_register(contest,user):
```

Comment:
Well, you might have some misunderstanding..
I mean
```
open_register = contest.open_register
if not open_register:
    # logger
    return False
has_ownership = user_info.has_contest_ownership(user,contest)
if has_ownership:
    # logger
    return False
# other conditions...
```
In this way, we don't have to check ```has_ownership`` if a contest is not open register.